compiler/rustc_codegen_ssa/src/debuginfo/type_names.rs

```diff
@@ -480,14 +482,34 @@ fn push_generic_params_internal<'tcx>(
                 push_debuginfo_type_name(tcx, type_parameter, true, output, visited);
                 output.push_str(", ");
             }
-            GenericArgKind::Const(const_parameter) => match const_parameter.val {
+            GenericArgKind::Const(ct) => match ct.val {
                 ty::ConstKind::Param(param) => write!(output, "{}, ", param.name).unwrap(),
-                _ => write!(
-                    output,
-                    "0x{:x}, ",
-                    const_parameter.eval_bits(tcx, ty::ParamEnv::reveal_all(), const_parameter.ty)
-                )
-                .unwrap(),
+                _ => {
+                    let bits_or_hash =
+                        match ct.try_eval_bits(tcx, ty::ParamEnv::reveal_all(), ct.ty) {
```

Comment:
Note that this will convert a lot of things to an integer representation, not just integers. So this will probably return `Some` when your type is actually some wrapper struct or a single element array or sth.

Reply:
Good to know! Maybe it would be better to emit the hash in those cases.